### ISSUE TITLE
Error while uploading an empty file to blob using azure-storage-blob: Data should be bytes or a seekable file-like object.

### ISSUE THREAD

**Issue by harneetvirk** [Azure/azure-sdk-for-python#25489]
- **azure-storage-blob**: 
- **12.13.0**: 
- **Windows**:
- **Python 3.8**: 

**Describe the bug**
While using azure-storage-blob 12.13.0, we are getting an exception on uploading an empty file to blob storage: ValueError: Data should be bytes or a seekable file-like object.

**To Reproduce**
1. conda create -n storageblobpy38 python=3.8 -y
2. conda activate storageblobpy38
3. pip install azure-storage-blob==12.13.0
4. python scratch2.py

Content of scratch2.py
```
from azure.storage.blob import BlobServiceClient
import os

def write_script(script_path, content):
    """
    Util for generating a python script, currently writes the file to disk.
    """
    with open(script_path, "w") as stream:
        stream.write(content)

write_script('train.py', "")

with open('train.py', 'rb') as data:
    account_name = 'SOME_ACCOUNT_NAME'
    account_key = 'SOMEACCOUNTKEY'
    account_url = "https://{account_name}.blob.{endpoint}".format(
        account_name=account_name, endpoint='core.windows.net'
    )
    client = BlobServiceClient(
        account_url=account_url,
        credential={
            "account_name": account_name,
            "account_key": account_key
        }
    )

    client.get_blob_client(
        container='SOMECONTAINER',
        blob='3.py'
    ).upload_blob(data=data, overwrite=True, validate_content=True)

```

**Expected behavior**
The empty file should be allowed to be uploaded on the blob.

**Screenshots**
Screenshot showing the error and stack trace:

![image](https://user-images.githubusercontent.com/20194110/182202633-3723d117-121f-4813-9e38-95ba0d4a71a0.png)

**Additional context**
An Empty file was allowed to be uploaded in the previous versions of azure-storage-blob

**Comment by Stevenjin8** [comment]
I think the issue is in

https://github.com/Azure/azure-sdk-for-python/blob/93a12a966d09c86c73386de531a121b21dd83232/sdk/core/azure-core/azure/core/rest/_rest_py3.py#L151-L158

I was able to solve it by changing the method to

```python
return self._data if self._data is not None else self._files
```

or by switching the order to

```python
return self._files or self._data
```

though the former seems more robust.

**Comment by Stevenjin8** [comment]
Spoke to @iscai-msft who is wrote the code above. She asked me to make a PR with the `None`-check

**Comment by jalauzon-msft** [comment]
This has been fixed in Storage by #25655, though the Core change in #25540 should still be considered.
Closing this issue since its resolved for Storage.
